cmd/query/app/querysvc/query_service.go

```diff
@@ -61,7 +63,11 @@ func NewQueryService(spanReader spanstore.Reader, dependencyReader dependencysto
 
 // GetTrace is the queryService implementation of spanstore.Reader.GetTrace
 func (qs QueryService) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
-	trace, err := qs.spanReader.GetTrace(ctx, traceID)
+	spanReader, err := factoryadapter.GetV1Reader(qs.traceReader)
```

Comment:
arguably, you could just call the tracer reader directly and transform the results into model - almost the same amount of code changes, but be done with it once and for all, no need to come back and revisit.

Reply:
@yurishkuro what would this look like? we haven't implemented the v2 reader interface for the factoryadapter at the moment so how would we do this?

Reply:
ah, ok, I thought we did. We need to, but can be different PR.